components/cop_codegen/src/rpn_function.rs

```diff
@@ -225,6 +229,77 @@ macro_rules! destruct {
     };
 }
 
+macro_rules! destruct2 {
+    ($elem:expr, $tp:ident::$var:ident, $msg:expr, $span_elem:expr) => {
+        if let $tp::$var(elem) = $elem {
+            elem
+        } else {
+            return Some($span_elem.span().unwrap().error($msg));
+        }
+    };
+    ($elem:expr, $tp:ident::$var:ident, $msg:expr) => {
+        destruct2!($elem, $tp::$var, $msg, $elem);
+    };
+}
+
+fn check_attr(meta: &[Ident], item_fn: &ItemFn) -> Option<Diagnostic> {
```

Comment:
Why not just use `Result<()>` as the return type? Then we don't need `destruct2`